org.eclipse.winery.frontends/app/topologymodeler/src/app/services/che.service.ts

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
+import { WineryRepositoryConfigurationService } from '../../../../tosca-management/src/app/wineryFeatureToggleModule/WineryRepositoryConfiguration.service';
+import { map } from 'rxjs/operators';
+import { ToastrService } from 'ngx-toastr';
+
+export interface CheResponse {
+    url: string;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class CheService {
+
+    constructor(private http: HttpClient,
+                private toastrService: ToastrService,
+                private configurationService: WineryRepositoryConfigurationService) {
+    }
+
+    getCheTheiaUrl(backendUrl: String): Observable<String> {
+        return this.http.get<CheResponse>(`${backendUrl}/admin/che`).pipe(map((res) => res.url));
+    }
+
+    async openChe(backendUrl: String, id: String, namespace: String, type: String) {
+        try {
+            const theiaUrl = await this.getCheTheiaUrl(backendUrl).toPromise();
+            const repositoryConfiguration = await this.configurationService.getRepositoryConfiguration().toPromise();
+            // TODO split this or just give theia the full path?
+            const pathParts = repositoryConfiguration.repositoryRoot.split('/');
+            const path = pathParts[2];
+            // TODO give full path as folderPath
+            const pathToOpen = theiaUrl + `/?ns=${namespace}&id=${id}&parentPath=${path}/${type}#/projects`;
```

Comment:
Here we need to resolve the TODOs. Also, I remember we decided to use the "full url" instead of parts.